### ISSUE TITLE
tf.data.Dataset core dump error when downloading mnist data set

### ISSUE THREAD

**Issue by hshankar61** [tensorflow/datasets#2930]
/!\ IF YOU WANT PEOPLE TO HELP YOU, PLEASE GIVE AS MUCH DETAIL AS POSSIBLE, INCLUDING THE FULL STACKTRACE AND CODE SNIPPET

**Short description**
Core dump when down loading the mnist data set

**Environment information**
* Operating System: Ubuntu
* Python version: 3.7.3
* `tensorflow-datasets`/`tfds-nightly` version: 4.2.0/4.2.0+nightly
* `tensorflow`/`tf-nightly` version: 2.4.0

* Does the issue still exists with the last `tfds-nightly` package (`pip install --upgrade tfds-nightly`) ? Yes

**Reproduction instructions**

```
import tensorflow as tf
import tensorflow_datasets as tfds
import numpy as np
import matplotlib.pyplot as plt

print(tf.__version__)
print(tfds.__version__)

mnist = tfds.builder('mnist')
mnist.download_and_prepare()
```
**Error output**
Dl Completed...: 100%|████████████████████████████████████████████████████████████| 4/4 [00:01<00:00,  2.33 file/s]
Dl Completed...: 100%|████████████████████████████████████████████████████████████| 4/4 [00:01<00:00,  2.76 file/s]
Segmentation fault (core dumped)

**Link to logs**
If applicable, <link to gist with logs, stack trace>

**Expected behavior**
Expect the mnist data set to be down loaded

**Additional context**
The same code works for the cats_or_dogs data set and hence it is probably not a local installation issue

cats_vs_dogs = tfds.builder('cats_vs_dogs')
cats_vs_dogs.download_and_prepare()

**Comment by rohit11544** [comment]
The issue is with the version of  tf-nightly. Install tf-nightly with version  **4.1.0+nightly** and try it. Even then if it fails to work then run it in  python **3.8.x** and use tf-nightly version as  **4.1.0+nightly** then it will surly work.


![Screenshot (32)](https://user-images.githubusercontent.com/61901749/104112597-b0786c00-5316-11eb-87ac-50bc004de4d1.png)
this fixes #2930

**Comment by vijayphoenix** [comment]
Unable to reproduce this issue.
It was probably a lack of system resources on your end which caused this.
Perhaps try it without other applications running simultaneously.
